pkg/services/ngalert/api/api_ruler_test.go

```diff
@@ -603,10 +604,19 @@ func createService(store *fakes.RuleStore) *RulerSrv {
 		cfg: &setting.UnifiedAlertingSettings{
 			BaseInterval: 10 * time.Second,
 		},
-		authz: accesscontrol.NewRuleService(acimpl.ProvideAccessControl(setting.NewCfg())),
+		authz:          accesscontrol.NewRuleService(acimpl.ProvideAccessControl(setting.NewCfg())),
+		amRefresher:    &fakeAMRefresher{},
+		featureManager: &featuremgmt.FeatureManager{},
 	}
 }
 
+type fakeAMRefresher struct {
+}
+
+func (f *fakeAMRefresher) RefreshConfig(_ context.Context, _ int64) error {
+	return nil
+}
```

Comment:
TDB: We need to add a test that asserts that whenever the notification settings are changed, the refresher gets called.